src/scenes/OrgRouter/NavBar.tsx

```diff
@@ -109,6 +114,31 @@ export default function NavBar({
     }
   }, [selectedOrganization]);
 
+  useEffect(() => {
+    const checkHasModules = async () => {
+      const projectsResult = await ProjectsService.listProjects(selectedOrganization.id);
```

Comment:
Can you use redux for this?

Reply:
I think the only addition to redux would be a new selector that takes a list of project IDs and returns the list of modules for those projects.

```
export const selectModuleLists = (projectIds: number[]) => (state: RootState) => projectIds.map((projectId) => state.moduleList[projectId]?.data);
```

Then you would just hook up the dispatch and selector for projects, `requestProjects` and `selectProjects`, and you'd have to dispatch the requests for the individual module lists too. Saying all that, it seems like perhaps this would all be better as its own function in one of the services that handles everything you have below. It just seems a bit heavy handed to have all of this within a component, let alone a component called "nav bar". Ideally this stuff is isolated elsewhere and the usage within this component is as simple as dispatch/select.

Reply:
Done! Could you review to see if my approach makes sense?

Reply:
I am wondering if I am using the feature flag correctly as well.

Reply:
Looks good, thanks for changing that up! A couple small suggestions.